### PR TITLE
[docs] Add a page on triggering EAS builds on the CI

### PR DESCRIPTION
**Pull request by satya164**
The page reuses text and examples from `guides/setting-up-continuous-integration.md`. I have also added additional example for GitHub actions.

I tested the CI workflow on GitHub actions and ensured that it triggers builds. I haven't tested on other CIs, but it should be the same as running `publish`.

### CODE REVIEW

File: docs/pages/build/building-on-ci.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+Next, we will configure the publishing step of your application to Expo.
+Before we can do this, we need to authenticate as the owner of the app.
+This is possible by storing the username or email with the password in the environment.
+Every vendor has implemented their storage mechanism for sensitive data like passwords, although they are very similar.
+Most vendors make use of environment variables which are "injected" into the environment and scrubbed from the logs to keep it safe.
+
+To perform the authentication, we will add this script to our configuration:
+
+```sh
+npx expo login -u <EXPO USERNAME> -p <EXPO PASSWORD>
```
Comment (wkozyra95) [inline docs/pages/build/building-on-ci.md:46]: We shouldn't recommend login/password auth on CI, `EXPO_TOKEN` is currently the recommended way

File: docs/pages/build/building-on-ci.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
+---
+title: EAS builds on the CI
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:2]:
```suggestion
title: EAS builds with CI
```

File: docs/pages/build/building-on-ci.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+---
+
+This document outlines how to trigger builds on EAS for your app from a CI environment.
+
+## Configuring the app
+
+To be able to trigger builds on EAS from the CI environment, first, we need to ensure that our app is configured to build on EAS and that we can trigger builds from our local machine.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:9]:
```suggestion
To trigger EAS builds from a CI environment, we first need to configure our app with EAS Build and make sure we can trigger builds from our local machine.
```

File: docs/pages/build/building-on-ci.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+To automatically configure your native project for building with EAS Build on Android and iOS, you will need to run the following command in the root of your project:
+
+```sh
+expo eas:build:init
+```
+
+See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for detailed explanation and examples of what this does.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:16]:
```suggestion
See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for a detailed explanation and examples of what this does.
```

File: docs/pages/build/building-on-ci.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+## Triggering builds
+
+To start triggering EAS builds on the CI, we need to install and configure `expo-cli` and then trigger new builds with the `eas:build` command.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:20]:
```suggestion
Before building with EAS on CI, we need to install and configure `expo-cli`. Then, we can trigger new builds with the `eas:build` command.
```

File: docs/pages/build/building-on-ci.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+### Prepare Expo CLI
+
+To interact with the Expo API, we need to install the Expo CLI.
+You can use an environment with this library preinstalled, or you can add it to the project as a development dependency.
+The latter is the easiest way but might increase the installation time.
+For vendors that charge you per minute, it might we worth creating a prebuilt environment.
+
+To install the Expo CLI into your project, you can execute this script.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:29]:
```suggestion
To install Expo CLI in your project, run: 
```

File: docs/pages/build/building-on-ci.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+```sh
+npm install --save-dev expo-cli
+```
+
+### Prepare authentication
+
+Next, we will configure the publishing step of your application to Expo.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:37]: I'm a little confused by this line- running `expo publish` isn't included in this guide?

File: docs/pages/build/building-on-ci.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+
+If you don't want to expose the password in the login script, set the `EXPO_CLI_PASSWORD` environment variable to the password and run the following script instead:
+
+```sh
+npx expo login --non-interactive -u <EXPO USERNAME>
+```
+
+### Trigger new builds
+
+After having the CLI library and authentication in place, we can finally create the build step.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:57]:
```suggestion
Now that we're authenticated with Expo CLI, we can create the build step.
```

File: docs/pages/build/building-on-ci.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+
+See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for detailed explanation and examples of what this does.
+
+## Triggering builds
+
+To start triggering EAS builds on the CI, we need to install and configure `expo-cli` and then trigger new builds with the `eas:build` command.
+
+### Prepare Expo CLI
+
+To interact with the Expo API, we need to install the Expo CLI.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:24]:
```suggestion
To interact with the Expo API, we need to install Expo CLI.
```

File: docs/pages/build/building-on-ci.md (hunk 1 above)
Comment (cruzach) [reply docs/pages/build/building-on-ci.md:46]: we can link them here for that- https://docs.expo.io/accounts/personal/#programmatic-access

**Review by dsokal — APPROVED** [review]
🚀

**Comment by satya164** [comment]
I don't have permissions to merge, so feel free to merge if it looks okay :)

File: docs/pages/build/building-on-ci.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+
+## Triggering builds
+
+Before building with EAS on CI, we need to install and configure `expo-cli`. Then, we can trigger new builds with the `eas:build` command.
+
+### Prepare Expo CLI
+
+To interact with the Expo API, we need to install Expo CLI. You can use an environment with this library preinstalled, or you can add it to the project as a development dependency.
+
+The later is the easiest way but might increase the installation time.
```
Comment (cruzach) [inline docs/pages/build/building-on-ci.md:26]:
```suggestion
The latter is the easiest way, but may increase the installation time.
```

**Pull request merged**